internal/gatewayapi/resource/load.go

```
@@ -70,13 +72,17 @@ func loadKubernetesYAMLToResources(input []byte, addMissingResources bool) (*Res
 
 		// Perform local validation and apply default values for gateway-api related resources only.
 		if gvk.Group == egv1a1.GroupName || gvk.Group == gwapiv1.GroupName {
-			if err = defaultValidator.Validate(yamlByte); err != nil {
-				return fmt.Errorf("local validation error: %w", err)
+			if validator != nil {
+				if err = validator.Validate(yamlByte); err != nil {
+					return fmt.Errorf("local validation error: %w", err)
+				}
```

can we return an err if validator is nil

and same for defaulter